pkg/backoff/backoff_test.go

```diff
@@ -38,28 +40,76 @@ func TestWithBackoff(t *testing.T) {
 
 	_, err = handle([]heartbeat.Heartbeat{})
 	require.NoError(t, err)
+
+	err = ini.ReadInConfig(v, tmpFile.Name())
+	require.NoError(t, err)
+
+	// make sure backoff settings were not written
+	assert.Empty(t, v.GetString("internal.backoff_at"))
+	assert.Empty(t, v.GetString("internal.backoff_retries"))
 }
 
 func TestWithBackoff_BeforeNextBackoff(t *testing.T) {
```

Comment:
This test was written to test the heartbeat won't be send during backoff, but this changes has changed it a lot and does not represent the original idea. Would be better to write another test to test backoff is written in the file. I also consider this kind of test more related to integration one.